Add API endpoint classes

atygesen opened the pull request.

Valleo4 commented on dupla/excpetions.py:
```diff
@@ -0,0 +1,13 @@
+__all__ = ["DuplaApiException", "DuplaApiAuthenticationException", "InvalidPayloadException"]
+
+
+class DuplaApiException(Exception):
+    pass
+
+
+class DuplaApiAuthenticationException(Exception):
+    pass
+
+
+class InvalidPayloadException(Exception):
+    """Current payload is invalid for the payload schema."""
```
file name has a typo :)

Valleo4 replied:
maybe add a small comment like the exception you added

Valleo4 commented on dupla/verification.py:
```diff
@@ -0,0 +1,23 @@
+from typing import Union
+from datetime import datetime, date
+
+ISO_FMT = "%Y-%m-%d"  # Date format for fields requiring yyyy-mm-dd
+
+
+def convert_and_validate_iso_date(value: Union[str, date, datetime]) -> str:
+    """Convert a date/datetime object to ISO.
+    The value is validated to be a valid yyyy-mm-dd date string."""
+    if isinstance(value, (date, datetime)):
+        value = value.strftime(ISO_FMT)
+    if not validate_iso_date(value):
+        raise ValueError(f"Date '{value}' is not a date of the format yyyy-mm-dd")
+    return value
+
+
+def validate_iso_date(date: str) -> bool:
+    """Verify that a date string has a valid correct yyyy-mm-dd format"""
+    try:
+        datetime.strptime(date, ISO_FMT)
+    except ValueError:
+        return False
+    return True
```
i would call this file validation instead of, dont know why but verification sound more authentication related. also since you named the utils validate_xx :)

atygesen replied:
Good point - `validate.py` it is.

Valleo4 commented on scripts/README.md:
```diff
@@ -0,0 +1 @@
+TODO
```
pls do :D

atygesen replied:
Done :)

atygesen commented:
Renamed `scripts/test_endpoints.py` to `scripts/run_endpoints.py` to avoid triggering pytest when running from the root folder.

Added a short `README.md` file describing how to use the provided `run_endpoints.py` testing script.

Valleo4 commented on scripts/README.md:
```diff
@@ -0,0 +1,48 @@
+`.env` and `agreements.yml` files. Below is a description of the variables
+in the `.env` file:
+
+| Variable          | Description                                                                                                                                                                                 |
+|-------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| DUBLA_API_URL     | The base URL to the SKAT API, such as `https://api.skat.dk` or `https://api.tfe.skat.dk`.                                                                                                   |
+| DUBLA_CERT_FILE   | Location to the certificate file for the "billetautomat", e.g. `~/certs/my_dupla_certificate.p12`.                                                                                          |
+| DUBLA_PASSWORD    | The password to the certificate file.                                                                                                                                                       |
+| BILLETAUTOMAT_URL | The URL to the "billetautomat", e.g. `https://oces.billetautomat.skat.dk/auth/realms/oces/certificates/cert` or `https://oces.billetautomat-uat.skat.dk/auth/realms/oces/certificates/cert` |
```
DUPLA* :)

atygesen replied:
Fixed :)

Valleo4 reviewed (approved):
looks good!
